faunadb-java/src/main/java/com/faunadb/client/FaunaClient.java

```diff
@@ -227,6 +229,15 @@ private Value handleResponse(FullHttpResponse response) {
       handleQueryErrors(response);
       JsonNode responseBody = parseResponseBody(response);
       JsonNode resource = responseBody.get("resource");
+
+      if(resource == null) {
+        throw new IOException("Invalid JSON.");
+      }
```

Comment:
when is this happening?

Reply:
If the server responds with a successful response, it should not happen. If the response does not have a `resource` field, it means the response is not successful. So, instead of swallowing the error with a `null` result, we should be throwing it.

This is the same as here:

https://github.com/fauna/faunadb-jvm/blob/4bc1ccd444b97ea5675a5645b3b6e6e5e3ea5749/faunadb-java/src/main/java/com/faunadb/client/FaunaClient.java#L316